contracts/libraries/LibBonds.sol

```diff
@@ -40,77 +42,77 @@ library LibBonds {
+    ///   - locks this amount from archaeologist's free bond; increases cursedBond by same
+    ///   - Adds the sarcophagus' id to the archaeologist's record of bonded sarcophagi
+    /// @param sarcoId Id of the sarcophagus with which to curse the archaeologist
+    /// @param archaeologist The archaologist to curse, with associated parameters of the curse
+    ///
+    /// @return the amount of digging fees due the embalmer for this curse
+    function curseArchaeologist(
+        bytes32 sarcoId,
+        EmbalmerFacet.CurseParams calldata archaeologist,
+        uint256 index
+    ) internal returns (uint256) {
         AppStorage storage s = LibAppStorage.getAppStorage();
+        LibTypes.Sarcophagus storage sarcophagus = s.sarcophagi[sarcoId];
 
-        // Increase the cursed bond amount
-        s.archaeologistProfiles[archaeologist].cursedBond += amount;
-    }
+        sarcophagus.cursedArchaeologists[archaeologist.archAddress] = LibTypes.CursedArchaeologist({
+            publicKey: archaeologist.publicKey,
+            privateKey: 0,
+            isAccused: false,
+            diggingFeePerSecond: archaeologist.diggingFeePerSecond
+        });
+        sarcophagus.cursedArchaeologistAddresses[index] = archaeologist.archAddress;
 
-    /// @notice Locks up the archaeologist's bond, decreasing the
-    /// archaeologist's free bond by an amount and increasing the
-    /// archaeologist's cursed bond by the same amount.
-    /// @param archaeologist The address of the archaeologist
-    /// @param amount The amount to lock up
-    function lockUpBond(address archaeologist, uint256 amount) internal {
-        // Decrease the free bond amount
-        decreaseFreeBond(archaeologist, amount);
+        // Calculate digging fees due for this time period (creationTime/previousRewrapTime -> resurrectionTime)
+        uint256 diggingFeesDue = archaeologist.diggingFeePerSecond *
+            (sarcophagus.resurrectionTime - sarcophagus.previousRewrapTime);
 
-        // Increase the cursed bond amount
-        increaseCursedBond(archaeologist, amount);
-    }
+        decreaseFreeBond(archaeologist.archAddress, diggingFeesDue);
+        s.archaeologistProfiles[archaeologist.archAddress].cursedBond += diggingFeesDue;
 
-    /// @notice Unlocks the archaeologist's bond, increasing the
-    /// archaeologist's free bond by an amount and decreasing the
-    /// archaeologist's cursed bond by the same amount.
-    /// @param archaeologist The address of the archaeologist
-    /// @param amount The amount to unlock
-    function unlockBond(address archaeologist, uint256 amount) internal {
-        // Decrease the cursed bond amount
-        decreaseCursedBond(archaeologist, amount);
+        s.archaeologistSarcophagi[archaeologist.archAddress].push(sarcoId);
 
-        // Increase the free bond amount
-        increaseFreeBond(archaeologist, amount);
+        return diggingFeesDue;
```

Comment:
Is there a reason to return `diggingFeesDue`?

Reply:
Yep -- digging fees calculation has been moved to this curse function, so the result is returned so that `createSarcophagus` can track the *total* digging fees due all archaeologists